controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    steering_controller:
      type: position_controllers/JointGroupPositionController
    
    velocity_controllers:
      type: velocity_controllers/JointGroupVelocityController

velocity_controllers:
  ros__parameters:
    use_sim_time: true
    joints: 
      - front_left_wheel_joint
      - front_right_wheel_joint
      - back_left_wheel_joint
      - back_right_wheel_joint
    command_interface: 
      - velocity
    state_interface: 
      - velocity
      - effort

steering_controller:
  ros__parameters:
    joints:
      - front_left_steering_joint
      - front_right_steering_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity

    state_publish_rate: 50.0 # Defaults to 50
    action_monitor_rate: 20.0 # Defaults to 20

    allow_partial_joints_goal: false # Defaults to false
    open_loop_control: true
    allow_integration_in_goal_trajectories: true
    constraints:
      stopped_velocity_tolerance: 0.01 # Defaults to 0.01
      goal_time: 0.0 # Defaults to 0.0 (start immediately)
